### PR TITLE
Move CI to GitHub Actions

### PR DESCRIPTION
**Pull request by samuell**
# Description

This PR moves the CI workflow configuration from the previous Travis CI one to GitHub actions, as suggested in #174

Some more details of changes done:

- The install script uses micromamba instead of conda if inside GitHub actions
- Some dependency specs were adjusted to make the installation go through
- A pointer to a folder `${CONDA_PREFIX}/testproject` was changed to `${CONDA_PREFIX}/testdata` instead, since the former does not exist in the repo.
- Two smaller bugs identified by flake8 was fixed, along with a typo

You can check the output of the converted code on the [actions tab of this repo](https://github.com/Clinical-Genomics/microSALT/actions).

## Primary function of PR
- [ ] Hotfix
- [ ] Patch
- [x] Minor functionality improvement
- [ ] New type of analysis
- [ ] Backward-breaking functionality improvement
- [ ] This change requires internal documents to be updated
- [ ] This change requires another repository to be updated

# Testing

_If the update is a hotfix, it is sufficient to rely on the development testing along with the Travis self-test automatically applied to the PR._

_Test routine to verify the stability of the PR:_
- _`bash /home/proj/production/servers/resources/hasta.scilifelab.se/install-microsalt-stage.sh BRANCHNAME`_
- _`us`_
- _`conda activate S_microSALT`_
- _(SITUATIONAL) `export MICROSALT_CONFIG=/home/proj/dropbox/microSALT.json`_
- _Select a relevant subset of the following:_
- _`microSALT analyse project MIC3109`_
- _`microSALT analyse project MIC4107`_
- _`microSALT analyse project MIC4109`_
- _`microSALT analyse project ACC5551`_

_Verify that the results for projects MIC3109, MIC4107, MIC4109 & ACC5551 are consistent with the results attached to AMSystem doc 1490, Microbial_WGS.xlsx_

## Test results
_These are the results of the tests, and necessary conclusions, that prove the stability of the PR._

# Sign-offs
- [ ] Code tested by @octocat
- [ ] Approved to run at Clinical-Genomics by @sylvinite

### CODE REVIEW

**Comment by samuell** [comment]
I haven't successfully tested the installation on Hasta yet, since it seems to break on some unrelated things (can not find databases etc).

I might need to discuss with some of you on how to best test this on Hasta.

File: microSALT/cli.py [hunk 1]
```diff
@@ -43,7 +43,8 @@
     sys.exit(-1)
 
 
-def set_cli_config(config):
+@click.pass_context
+def set_cli_config(ctx, config):
```
Comment (Vince-janv) [inline microSALT/cli.py:47]: Was this function broken previously? Where did it get ctx from?

**Review by Vince-janv — APPROVED** [review]
Nice 👍

File: microSALT/cli.py (hunk 1 above)
Comment (samuell) [reply microSALT/cli.py:47]: It apparently didn't get it from anywhere. Flake8 complained that it was undefined, and I haven't been able to find anything else (fixture-like) that would explain the definition. Though, I might be missing if something "magical" is going on here!

**Pull request merged**